frontend/src/components/ProjectEditor.vue

```diff
@@ -311,21 +351,95 @@ const updateFrameWeight = (frame: Frame) => {
       }
     }
   }
-  // Log the updated frame weight (optional)
   console.log(`Frame ${frame.frameNumber} weight updated to ${frame.weight}`);
 };
 
+const loadProjectData = async () => {
+  if (!uploadedVideoUUID.value) return;
+  try {
+    const response = await axios.get(`${BACKEND_URL}/projects/${uploadedVideoUUID.value}`);
+    if (response.status != 200) {
+      console.error('Error fetching projectData:', response);
+      return;
+    }
+    const projectData: ProjectDataResponse = response.data;
+
+    scale.value = projectData.scale;
+    framesPerSecond.value = projectData.fps;
+
+    // If there's a long exposure image, show it
+    if (projectData.latest_long_exposure_image_name) {
+      longExposureImageUrl.value = projectData.latest_long_exposure_image_name;
+      showTimeline.value = false
+    }
+
+    // Load Video from Backend
+
+    videoUrlOnServer.value = `${BACKEND_URL}/uploads/${uploadedVideoUUID.value}.${projectData.video_file_extension}`;
+
+  } catch (error) {
+    console.error('Error loading project data:', error);
+  }
+};
+
+const loadTimelineThumbnails = async () => {
+  if (!uploadedVideoUUID.value || !framesPerSecond.value) {
+    return
+  }
+
+  const allFramesArr: Frame[] = [];
+  const duration = uploadedVideoDuration.value
+
+  for (let i = 1; i <= Math.ceil(duration * framesPerSecond.value); i++) {
+    allFramesArr.push({
+      src: `${BACKEND_URL}/outputs/${uploadedVideoUUID.value}/frames/ffout_thumbnail_${i.toString().padStart(4, '0')}.webp`,
```

Comment:
Ich würd das aufbauen von ner URL die aufs eigene Backend zeigt evtl auslagern in nen eigenen Helper. Muss nicht direkt ne Klasse sein, aber ich habs immer gerne wenn Envs oder feste Pfade nur ein einer Stelle gebastelt werden. Wenn sich irgendwann die Api ändert muss man sonst überall im Code nach magischen Strings suchen.

Hier an der Stelle zugegeben etwas eklig einzubauen, einfach weil ein bissl am Anfang und ein bissl am Ende des Strings dynamisch ist, aber grundsätzlich ist die Idee von nem `Api` helper mit simplen Funktionen für feste Endpunkte ne gute Idee. Dann kann man auch mit Code Generation Arbeiten (falls das Backend es unterstützt) für "End to End" Typesafety